awx/main/tasks.py

```diff
@@ -2120,7 +2134,25 @@ def _build_scm_url_extra_vars(self, project_update):
         return scm_url, extra_vars
 
     def build_inventory(self, instance, private_data_dir):
-        return 'localhost,'
+        data = {
+            "_meta": {
+                "hostvars": {}
+            },
+            "all": {
+                "hosts": ["localhost"]
+            }
+        }
+        json_data = json.dumps(data)
+        runner_inventory_folder = os.path.join(private_data_dir, 'inventory')
+        fn = os.path.join(runner_inventory_folder, 'hosts')
+        if not os.path.exists(runner_inventory_folder):
+            # Ansible Runner requires that this directory exists.
+            # Specifically, when using process isolation
+            os.mkdir(runner_inventory_folder)
+        with open(fn, 'w') as f:
+            os.chmod(fn, stat.S_IRUSR | stat.S_IXUSR | stat.S_IWUSR)
+            f.write('#! /usr/bin/env python3\n# -*- coding: utf-8 -*-\nprint(%r)\n' % json_data)
+        return fn
```

Comment:
I'd like to figure out how to do this more cleanly. In my own testing I've had pretty good luck just not including any hosts, because `ansible-playbook` doesn't need `-i` to work, and it automatically adds localhost.

Reply:
This has been addressed.